### PR TITLE
feat: Messaging

### PR DESCRIPTION
**Pull request by LennartKoot**

### CODE REVIEW

File: ShipitSmarter.Core.sln [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
+#
```
Comment (Droxx) [inline ShipitSmarter.Core.sln:3]: Is this a mistake by the IDE that needs cleaning up?

**Review by Droxx — COMMENTED** [review]
Looks super impressive, just one comment on the contents. I'm going to check out the example and see it in action

File: ShipitSmarter.Core.sln (hunk 1 above)
Comment (LennartKoot) [reply ShipitSmarter.Core.sln:3]: That might have been me accidentalily typing in here instead of a README.md, will fix :)

Comment (LennartKoot) [reply ShipitSmarter.Core.sln]: Added new projects again to the sln and this is re-added. So it seems intentional by running the `dotnet sln add` command.

File: test/ShipitSmarter.Core.Messaging.Receiver.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
+using FluentAssertions;
+using Google.Cloud.PubSub.V1;
+using Google.Protobuf;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace ShipitSmarter.Core.Messaging.Receiver.Tests;
+
+public class GoogleSubscriberClientTests
```
Comment (Droxx) [inline test/ShipitSmarter.Core.Messaging.Receiver.Tests/UnitTest1.cs:10]: FYI the filename does not match the class

**Review by Droxx — APPROVED** [review]
LGTM, just one misnamed file

**Pull request merged**